types/tedious/index.d.ts

```diff
@@ -372,6 +372,11 @@ export interface ConnectionAuthenticationOptions {
      */
     clientId?: string | undefined;
 
+    /**
+     * Client secret for the service principal identity
+     */
+    clientSecret?: string | undefined;
```

Comment:
this is not present in the code for tedious@4, which is what this version of tedious types.

Unfortunately, tedious@4 is 5 years old, so I suspect it's a lot of work to update these types to 16. In any case, tedious is now written in typescript, so you should instead ask the author to ship types in .d.ts alongside .js in the tedious package itself.